shared/video-sdk/develop/speech-to-text/project-implementation/android.mdx

```diff
@@ -0,0 +1,372 @@
+    android:layout_alignStart="@id/remote_video_view_container"
+    android:layout_alignEnd="@id/remote_video_view_container"
+    android:onClick="SpeechToText"
+    android:text="Start speech to text" />
+
+
+<Button
+    android:id="@+id/QuerySpeechToTextButton"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_below="@id/SpeechToTextButton"
+    android:layout_alignStart="@id/remote_video_view_container"
+    android:layout_alignEnd="@id/remote_video_view_container"
+    android:onClick="query"
+    android:text="Query speech to text" />
+```
+
+### Handle the system logic
+
+1. **Integrate a `HTTP` client library into your project**
+
+    To call `HTTP` methods that start, stop, and query a speech-to-text task, you add a `HTTP` client library to your project. Open `/Gradle Scripts/build.gradle (Module: <project name>.app)` and add the following line under `dependencies {`:
+
+    ```groovy
+    implementation 'com.squareup.okhttp3:okhttp:4.10.0'
+    ```
+
+1. **Import the required classes**
+
+    To use the relevant `okhttp` classes in your project, add the following statements after the last `import` statement in `/app/java/com.example.<projectname>/MainActivity`.
+
+    ```java
+    import okhttp3.Call;
+    import okhttp3.Callback;
+    import okhttp3.MediaType;
+    import okhttp3.OkHttpClient;
+    import okhttp3.Request;
+    import okhttp3.RequestBody;
+    import okhttp3.Response;
+
+    import java.io.IOException;
+    import java.util.Base64;
+
+    import org.json.JSONArray;
+    import org.json.JSONException;
+    import org.json.JSONObject;
+    ```
+
+1. **Define variables to manage a speech-to-text task**
+
+    In `/app/java/com.example.<projectname>/MainActivity`, add the following declarations to the `MainActivity` class:
+
+    ```java
+    private String baseUrl = "https://api.agora.io";
+    private String apiKey = "<your API key from Agora console>";
+    private String apiSecret = "<your API secret from Agora console>";
+    private String instanceID = "Test-1";
+    private String authorizationHeader, builderToken;
+    private String speechToTextTaskId = ""; // Returned by the start method
+    private boolean isSpeechToTextRunning = false;
+    private Button btnSpeechToText;
+    ```
+
+### Manage a speech-to-text task
+
+Take the following steps to start, stop and query a text-to-speech task in your <Vpl k="CLIENT" />.
+
+1. **Get a `builderToken`**
+
+    To use the speech to text RESTful API, you generate a Base64-encoded credential with the API key and secret provided by <Vg k="COMPANY" />. You add this credential to the request header as a parameter named `Authorization`. In the request body, you pass a parameter `instanceID` that you choose to identify the instance. To [acquire a `builderToken`](#acquire-a-buildertoken), add the following method to the `MainActivity` class:
+
+    ```java
+    public void getBuilderToken() throws IOException {
```

Comment:
Needs a `@RequiresApi(api = Build.VERSION_CODES.O)` before it according to my IDE

Reply:
Same as above.